src/app/Stopper.cpp

```diff
@@ -36,7 +37,7 @@ Stopper::~Stopper()
 void
 Stopper::setOnStop(std::function<void(boost::asio::yield_context)> cb)
 {
-    boost::asio::spawn(ctx_, std::move(cb));
+    boost::asio::spawn(ctx_, std::move(cb), boost::asio::detached);
```

Comment:
Using `detached` everywhere may (as far as I know) drop uncaught exceptions in coroutines. I think it is better to have crash when an exception we didn't expect was thrown.

Reply:
Right 😢 It is passing our tests but i guess we just did not check for exception propagation if/where needed. It's easier to preserve the old behaviour through a proxy function.